enterprise/cmd/frontend/internal/batches/resolvers/changeset.go

```diff
@@ -352,6 +353,22 @@ func (r *changesetResolver) ForkName() *string {
 	return nil
 }
 
+func (r *changesetResolver) CommitVerification(ctx context.Context) (graphqlbackend.CommitVerificationResolver, error) {
+	switch r.changeset.ExternalServiceType {
+	case extsvc.TypeGitHub:
+		r.spec, r.specErr = r.store.GetChangesetSpecByID(ctx, r.changeset.CurrentSpecID)
+		if r.specErr != nil {
+			return nil, r.specErr
+		}
+		if r.spec.CommitVerification != nil {
+			return &commitVerificationResolver{
+				commitVerification: r.spec.CommitVerification,
+			}, nil
+		}
```

Comment:
For this, I think we can use the `computeSpec` utility:

```suggestion
		spec, err := r.computeSpec(ctx)
		if err != nil {
			return nil, err
		}
		
		if spec.CommitVerification != nil {
			return &commitVerificationResolver{
				commitVerification: spec.CommitVerification,
			}, nil
		}
```

The utility uses a `sync.Once` cache to ensure that if a GraphQL request came in asking for multiple fields on a `Changeset` that would require us to look up the current changeset spec by ID in order to resolve, we only perform that work and make that trip to the DB one time across the whole request!

Reply:
Oh that's really cool!!